### PR TITLE
Create generator for mocha.opts file

### PR DESCRIPTION
**Pull request by riteshe63**
Add mocha Framework to test cases.

### CODE REVIEW

File: generators/test/templates/mocha.opts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+--reporter spec
+--ui bdd
+--recursive
+
```
Comment (gaurav21r) [inline generators/test/templates/mocha.opts]: @riteshe63 Why so many extra spaces?

File: generators/test/index.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+const yeoman = require('yeoman-generator');
```
Comment (gaurav21r) [inline generators/test/index.js:1]: @riteshe63 Is this file linted with eslint?

File: generators/test/index.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+const yosay = require('yosay');
+const _includes = require('lodash/includes');
+
+module.exports = yeoman.Base.extend({
+  prompting: {
+    mocha() {
+      const self = this;
+
+      this.log(yosay(
+        'Add Testing Framework for test cases'
```
Comment (gaurav21r) [inline generators/test/index.js]: I think it would be better to say **Installing Testing frameworks**

File: generators/test/index.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      ));
+      return this.prompt([{
+        type: 'checkbox',
+        name: 'selectedFramework',
```
Comment (gaurav21r) [inline generators/test/index.js]: @riteshe63 Framework is too generic. It needs to `selectedTestFramework`.

File: generators/test/index.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        default: ['Mocha'],
+        when: () => !self.config.get('selectedFramework'),
+      }])
+      .then(answers => {
+        self.selectedFramework = self.config.get('selectedFramework') || answers.selectedFramework;
+      });
+    },
+  },
+  /**
+   * Finally runs an npm install as well as installs plugins for editors.
```
Comment (gaurav21r) [inline generators/test/index.js]: @riteshe63 Irrelevent here as we are not installing anything for editors ATM.

File: generators/test/index.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+   */
+  writing() {
+    // Mocha
+    if (_includes(this.selectedFramework, 'Mocha')) {
+      this.log('Configuring Mocha for mocha test cases');
```
Comment (gaurav21r) [inline generators/test/index.js]: Should be _Configuring default options for mocha_

**Review by gaurav21r — CHANGES_REQUESTED** [review]

**Review by gaurav21r — CHANGES_REQUESTED** [review]
@riteshe63 Please ask @neerajsachan to submithis grunt file here.

**Comment by neerajsachan** [comment]
Actually, presently  we are testing all unit test in the test/spec/unit/**/*.js file path in mocha-Test.  Can we use  test/unit/**/_.js path instead of the test/spec/unit/__/_.js path ? .

**Comment by riteshe63** [comment]
@gaurav21r  All Review is done, Please check.

**Comment by neerajsachan** [comment]
Change path of the folder: generators/test/templates/grunt instead of generators/grunt .

**Comment by riteshe63** [comment]
@gaurav21r Please review code.

**Pull request merged**